### ISSUE TITLE
Use Weather API for Location-based Adaptive Brightness

### ISSUE THREAD

**Issue by sampittko** [alin23/Lunar#231]
### Name of the feature

Location-based Adaptive Brightness with Weather Conditions Consideration

### What problem does this feature solve?

If foggy or simply brightness of the environment at `location(lat, lng)` is lower at time `t` than predicted by the current algorithm, algorithm would get more accurate in case of current weather conditions consideration. Location-based Adaptive Brightness cannot adapt properly in the current state of the algorithm and manual adjustements are required in case of cloudy weather conditions that lower the brightness of the environment.

### How should the feature work

When fetching location, query some kind of weather API to get weather information at `location(lat, lng)` and calculate brightness based on the two informations (location, weather).

Basically if there are clouds during the daylight, lower the brightness even more.

**Comment by sampittko** [comment]
After research, I found that something similar has already been proposed inside #118

**Comment by alin23** [comment]
Yes, I'll close this in favor of #118 

The feature is WIP but I deprioritized it because I wanted to first make Lunar work on M1. Since that is impossible for the moment, I'll get back to the weather aware location mode feature.
